### PR TITLE
Refact NoticeMixin

### LINKED ISSUE #3424: Toast breaks when Indefinite Toast opens and closes while another toast is active.
### Overview of the problem
We encountered this issue on our app, and found out that it's also happening on the example on the website. This issue revolves around a regular Toast and an Indefinite Toast occurring almost at the same time. 

**Buefy** version: Latest
**Vuejs** version: [2.3.0]
**OS/Browser**: Windows/Edge

### Description
When a regular toast with some timeout (3000ms for example) is being displayed, and we display an Indefinite Toast and close it while the regular Toast is still active, it will break and won't allow us to create or open any new toast.

**This is reproducible with the current example on the website**

### Steps to reproduce
image
1. Navigate to the toast example https://buefy.org/documentation/toast/#from-inside-vue-instance
2. Hit the green `Launch Toast (custom)` button
3. While the toast is active, hit `Launch Toast (indefinite)` and then `Close Toast (indefinite)`, while the toast from step 2 is still active.
4. Observe that toast will stop working.

### Expected behavior
Toast to continue function as normal.

### Actual behavior
Toast stopped functioning. Cannot open new toast messages.

### PR DESCRIPTION
**Pull request by geeksilva97**
This PR updates the `NoticeMixin` and Fixes #3424

## Proposed Changes

- Remove the recursive call when showing Notices

## Snackbar behavior after refactoring
ezgif com-gif-maker

## Toast behavior after refactoring
ezgif com-gif-maker (1)

### CODE REVIEW

**Comment by jtommy** [comment]
PR is ok but changing the prop name it becomes a breaking change.
I prefer to fix/improve the code without introducing changes to dev's code

**Comment by geeksilva97** [comment]
> PR is ok but changing the prop name it becomes a breaking change. I prefer to fix/improve the code without introducing changes to dev's code

Hi, @jtommy. We can keep the variable names, I think that is OK. I changed it because I thought it could be clearer.

File: src/components/snackbar/index.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ const SnackbarProgrammatic = {
             delete params.message
         }
         const propsData = merge(defaultParam, params)
-        propsData.replace = true
+        propsData.queue = true
         const vm = typeof window !== 'undefined' && window.Vue ? window.Vue : localVueInstance || VueInstance
```
Comment (jtommy) [inline src/components/snackbar/index.js:33]: Are you forcing it to be true??

**Review by jtommy — CHANGES_REQUESTED** [review]

File: src/components/snackbar/index.js (hunk 1 above)
Comment (geeksilva97) [reply src/components/snackbar/index.js]: It was a mistake I made when  I tried to set `queue: true` as default in snackbars. I fixed it moving the `queue` property to the `defaultParams` object.

**Pull request merged**